srp_daemon/srp_daemon.c

```diff
@@ -1991,12 +2045,16 @@ static int ibsrpdm(int argc, char *argv[])
 		}
 	}
 
+	ret = setup_wakeup_fd();
```

Comment:
Yes, I'm sure that ibsrpdm doesn't need wakeup_pipe[] to be initialized and also that it would be an improvement if the signal actions are left unchanged for ibsrpdm. There are four calls to get_received_signal() in main() but these are all in code that cannot be reached when executed as ibsrpdm.